db/db_impl/db_impl.cc

```diff
@@ -4185,9 +4203,18 @@ void DBImpl::GetApproximateMemTableStats(ColumnFamilyHandle* column_family,
   ColumnFamilyData* cfd = cfh->cfd();
   SuperVersion* sv = GetAndRefSuperVersion(cfd);
 
+  const Comparator* const ucmp = column_family->GetComparator();
+  assert(ucmp);
+  size_t ts_sz = ucmp->timestamp_size();
+
+  Slice start, limit;
+  // Add timestamp if needed
+  std::string start_with_ts, limit_with_ts;
+  std::tie(start, limit) = MaybeAddTimestampsToRange(
```

Comment:
Now that we're on C++17, we could consider using structured bindings here and below in `GetApproximateSizes` (i.e. `auto [start, limit] = ...`).

Reply:
That's a good idea! Reminds me next time I consult Chatgpt for syntax, I should start with "with C++ 17, how to ...."😂